src/System.IO.FileSystem.AccessControl/tests/FileSystemAclExtensionsTests.cs

```diff
@@ -187,5 +190,102 @@ public void SetAccessControl_FileStream_FileSecurity_Success()
+        }
+
+        [Theory]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.FullControl, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.ReadAndExecute, AccessControlType.Allow)]
+        [InlineData(WellKnownSidType.BuiltinUsersSid, FileSystemRights.Write, AccessControlType.Allow)]
+        public void DirectoryInfo_Create_DirectorySecurityWithSpecificAccessRule(WellKnownSidType sid, FileSystemRights rights, AccessControlType controlType)
+        {
+            DirectorySecurity security = new DirectorySecurity();
+            SecurityIdentifier identity = new SecurityIdentifier(sid, null);
+            FileSystemAccessRule accessRule = new FileSystemAccessRule(identity, rights, controlType);
+
+            security.AddAccessRule(accessRule);
+
+            CreateAndVerifyDirectoryWithSecurity(security);
+        }
+
+        private void CreateAndVerifyDirectoryWithSecurity(DirectorySecurity security)
+        {
+            using var directory = new TempDirectory();
+
+            string path = Path.Combine(directory.Path, "directory");
+            DirectoryInfo info = new DirectoryInfo(path);
+
+            info.Create(security);
+
+            Assert.True(Directory.Exists(path));
+            Assert.Equal(typeof(FileSystemRights), security.AccessRightType);
+
+            DirectorySecurity actualSecurity = info.GetAccessControl();
+            Assert.Equal(typeof(FileSystemRights), actualSecurity.AccessRightType);
+
+            List<FileSystemAccessRule> expectedRules = security.GetAccessRules(includeExplicit: true, includeInherited: false, typeof(SecurityIdentifier))
+                .Cast<FileSystemAccessRule>().ToList();
+
+            List<FileSystemAccessRule> actualRules = actualSecurity.GetAccessRules(includeExplicit: true, includeInherited: false, typeof(SecurityIdentifier))
+                .Cast<FileSystemAccessRule>().ToList();
+
+            // If DirectorySecurity is created without arguments, GetAccessRules will return zero rules
+            Assert.Equal(expectedRules.Count, actualRules.Count);
+            if (expectedRules.Count > 0)
+            {
+                Assert.True(actualRules.Count > 0);
+
+                Assert.All(expectedRules, actualRule =>
+                {
+                    int count = expectedRules.Count(expectedRule => AreRulesEqual(expectedRule, actualRule));
+                    Assert.True(count > 0);
+                });
+            }
+        }
+
+        private bool AreRulesEqual(FileSystemAccessRule expectedRule, FileSystemAccessRule actualRule)
+        {
+            return
+                expectedRule.AccessControlType == actualRule.AccessControlType &&
+                expectedRule.FileSystemRights  == actualRule.FileSystemRights &&
+                expectedRule.InheritanceFlags  == actualRule.InheritanceFlags &&
+                expectedRule.PropagationFlags  == actualRule.PropagationFlags;
```

Comment:
what about `IsInherited`  and `IdentityReference` properties -- do they need comparing also?

Reply:
Looks like `IsInherited` will always be false.

Reply:
Let me check.